types/src/generator/structure.ts

```diff
@@ -0,0 +1,405 @@
+import assert from "assert";
```

Comment:
maybe this is asking too much, but having some simple tests for this could would be nice.

Test Json capnp => source code for small pieces? This will make maintaining the code easier.